post.md

```diff
@@ -0,0 +1,201 @@
+second requires specially crafted kernel. Kernel provided by default in
+`xen-linux-system-amd64` lacks NFSv3 built in module, what is necessary to
+mount NFS during boot.
+
+### Linux kernel compilation
+
+Kernel provided with my upgrade was oldstable `4.9.51`. You can check that in
+apu2 or NFS `/boot` directory. We have to recompile that kernel and enable
+NFSv3.
+
+```
+cd linux-stable-4.9.51
+cp /path/to/pxe-server/debian/debian-stable/boot/config-4.9.0-4-amd64 .config
+```
+
+Then edit config and mark `NFSv3` options as `y`:
+
+```
+CONFIG_NFS_FS=m
+CONFIG_NFS_V2=m
+CONFIG_NFS_V3=y
+CONFIG_NFS_V3_ACL=y
+CONFIG_NFS_V4=m
+CONFIG_NFS_SWAP=y
+CONFIG_NFS_V4_1=y
+CONFIG_NFS_V4_2=y
+```
+
+For conveninence we compiled `deb-pkg` version:
+
+```
+make -j$(nproc) deb-pkg
+```
+
+Let's copy and install results of our compilation:
+
+```
+
+```
+
+### Booting Xen
+
+Components needed for that cofiguration are:
+
+* `xen-4.8-amd64`, which can be get from
+  `pxe-server/debian/debian-stable/boot/xen-4.8-amd64.gz` and unpacked using
+`zcat xen-4.8-amd64.gz > xen-4.8-amd64`
+
+* ``, which can be get from `pxe-server/debian/debian-stable/boot/vmlinuz-`
+
+We have to adjust `pxe-server/netboot/menu.ipxe` which we use to boot whole syste. We
+played little bit with the options before we get to this configuration.
+
+First modify menu:
+
+```
+diff --git a/menu.ipxe b/menu.ipxe
+index e11e44ac13d3..a610bb2f20b7 100644
+--- a/menu.ipxe
++++ b/menu.ipxe
+@@ -5,6 +5,7 @@ menu
+ item --gap -- ---------------- iPXE boot menu ----------------
+ item shell          ipxe shell
+ item deb-netboot    Debian stable netboot
++item xen    Xen
+ item deb-stable-netinst    TODO:Debian stable netinst
+ item deb-testing-netinst    TODO:Debian testing netinst
+ item deb-testing-netinst-uefi    TODO:Debian testing netinst (UEFI-aware)
+```
+
+Then add blow `deb-netboot`:
+
+```
+:xen
+kernel xen-4.8-amd64 dom0_mem=512M loglvl=all guest_loglvl=all com1=115200,8n1 console=com1
+module console=hvc0 earlyprintk=xen nomodeset root=/dev/nfs rw ip=dhcp nfsroot=<NFS_SRV_IP>:/srv/nfs/debian/debian-stable,vers=3,udp nfsrootdebug
+boot
+goto MENU
+```
+
+Do not forget to replace `<NFS_SRV_IP>`.
+
+This should give us bootable dom0 na neat Xen logs very useful for debugging:
+
+```
+```
+
+# Xen in Qubes OS
+
+On my laptop Lenovo ThinkPad T460p with i7-6820HQ I see below problems in `xl dmesg` outout:
+
+* `No NUMA configuration found`
+    - it doesn't seem to make sense in single socket system, so let's not bother
+* `Couldn't initialize a 1920x1080 framebuffer early`
+* `ACPI: 32/64X FACS address mismatch in FADT - b7f69000/0000000000000000, using 32`
+* `ERST table was not found`
+* `Not enabling x2APIC (upon firmware request)`
+* `Intel VT-d Snoop Control not enabled.`
+* `Intel VT-d Dom0 DMA Passthrough not enabled.`
+* `Intel VT-d Posted Interrupt not enabled.`
+* `VPMU: disabled`
+* `[VT-D]Passed iommu=no-igfx option.  Disabling IGD VT-d engine.`
+
```

Comment:
Would be great to see some effect like PCI passthrough and some short guide how to do it. I believe it is on the TODO list.